### PR TITLE
arch: arm: nrf: Enable SEGGER RTT on all Nordic SoCs.

### PR DESCRIPTION
**Pull request by pizi-nordic**
All chips from nRF51 and nRF52 series supports Segger RTT,
so we can enable it if given series is used.

Signed-off-by: Piotr Zięcik <piotr.ziecik@nordicsemi.no>

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]
Yep, makes sense. Does not depend on the IC variant at all.

File: arch/arm/soc/nordic_nrf/nrf51/Kconfig.series [hunk 1]
```diff
@@ -17,5 +17,6 @@ config  SOC_SERIES_NRF51X
 	select XIP
 	select HAS_CMSIS
 	select HAS_NRFX
+	select HAS_SEGGER_RTT
```
Comment (ioannisg) [inline arch/arm/soc/nordic_nrf/nrf51/Kconfig.series]: This was missing from nRF51; just to double-check that this is the indented behavior, now.

**Review by anangl — APPROVED** [review]
LGTM, although you could make it (almost) perfect by removing the period from the commit title.

**Comment by ioannisg** [comment]
commit message: supports --> support

**Review by ioannisg — CHANGES_REQUESTED** [review]
OK. A request for clarification and a minor typo in commit message

**Comment by pizi-nordic** [comment]
@ioannisg: Comments addressed. Thank you.

**Review by ioannisg — APPROVED** [review]

**Comment by carlescufi** [comment]
@ioannisg @pizi-nordic actually let me just double-check: Segger RTT support has nothing to do with the SoC, does it? but rather with the debugger IC that comes with the board? Does it still make sense to move to the SoC's Kconfig?

**Comment by galak** [comment]
> @ioannisg @pizi-nordic actually let me just double-check: Segger RTT support has nothing to do with the SoC, does it? but rather with the debugger IC that comes with the board? Does it still make sense to move to the SoC's Kconfig?

I wondering if we should just remove the HAS_SEGGER_RTT, and let the user enable it if they desire?

**Comment by pizi-nordic** [comment]
@carlescufi: The chip must have debugging interface compatible with Segger (like SWD).
@galak: IMHO HAS_SEGGER_RTT should be set by SoC config (as not all SoCs supports Segger). However it's actual usage should be controlled by other switch (is part is missing AFAIK).

**Comment by ioannisg** [comment]
Then, it looks to me that the proper way to do this is to define generic SOC capabilities (if we do not have already), i.e. CPU_HAS_SWD. And then we should have BOARD_HAS_SEGGER_RTT depending on CPU_HAS_SWD, and defined as "default-y" in nRF5x boards.

Similar to what we did in #8650, @carlescufi ?

**Review by nashif — APPROVED** [review]

**Comment by carlescufi** [comment]
SWD is very generic and has really nothing to do with Segger AFAIK. I actually think that this was okayish in the first place. The actual Segger RTT code is only compiled if `CONFIG_RTT_CONSOLE` is enabled, so `CONFIG_HAS_SEGGER_RTT` is just there to make sure the board has a segger-compatible chip. That said some boards support external debuggers so I am leaning towards getting rid completely of `HAS_SEGGER_RTT` and simply continuing to use `RTT_CONSOLE` as the only condition to compile-in the RTT code.

Also RTT doesn't even seem to require SWD:
>SEGGER RTT can be used with any J-Link model and any supported target processor which allows background memory access, which are Cortex-M and RX target

As described [here](https://www.segger.com/products/debug-probes/j-link/technology/about-real-time-transfer/)

**Comment by carlescufi** [comment]
Adding @nordic-krch for additional input here, since @sjanc sees sizes increases when enabling `HAS_SEGGER_RTT` and that should not be the case IMHO

**Comment by carlescufi** [comment]
@ioannisg @pizi-nordic are you sure that you need SWD for RTT? I don't see that in the doc I linked above

**Comment by pizi-nordic** [comment]
Basically, the HAS_SEGGER_RTT means "is supported by SEGGER JLink". This does not mean, that chip has to have specific hardware. As all nRF5x support JLink, all of them support RTT. Similarly, the nRF24 does not support RTT as they are not supported by JLink.

**Review by nordic-krch — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**